redfish-core/lib/managers.hpp

```diff
@@ -1922,6 +1922,70 @@ inline void setDateTime(std::shared_ptr<bmcweb::AsyncResp> aResp,
     }
 }
 
+/**
+ * @brief Retrieves bmc state properties over dbus
+ *
+ * @param[in] aResp     Shared pointer for completing asynchronous calls.
+ *
+ * @return None.
+ */
+inline void getBMCState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
+{
+    crow::connections::systemBus->async_method_call(
+        [aResp](const boost::system::error_code ec,
+                const std::variant<std::string>& bmcState) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "DBUS response error " << ec;
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            const std::string* state = std::get_if<std::string>(&bmcState);
+            if (!state)
+            {
+                messages::internalError(aResp->res);
+                return;
+            }
+            if (*state == "xyz.openbmc_project.State.BMC.BMCState.Ready")
+            {
+                aResp->res.jsonValue["PowerState"] = "On";
+                aResp->res.jsonValue["Status"]["State"] = "Enabled";
+                aResp->res.jsonValue["Status"]["Health"] = "OK";
+            }
+            else if (*state == "xyz.openbmc_project.State.BMC.BMCState."
+                               "Quiesced")
+            {
+                aResp->res.jsonValue["PowerState"] = "On";
+                aResp->res.jsonValue["Status"]["State"] = "Quiesced";
+                aResp->res.jsonValue["Status"]["Health"] = "Critical";
+            }
+            else if (*state == "xyz.openbmc_project.State.BMC.BMCState."
+                               "NotReady")
+            {
+                aResp->res.jsonValue["PowerState"] = "PoweringOn";
+                aResp->res.jsonValue["Status"]["State"] = "Starting";
+                aResp->res.jsonValue["Status"]["Health"] = "OK";
+            }
+            else if (*state == "xyz.openbmc_project.State.BMC.BMCState."
+                               "UpdateInProgress")
+            {
+                aResp->res.jsonValue["PowerState"] = "PoweringOn";
+                aResp->res.jsonValue["Status"]["State"] = "Updating";
+                aResp->res.jsonValue["Status"]["Health"] = "OK";
+            }
+            else
+            {
+                aResp->res.jsonValue["PowerState"] = "Off";
```

Comment:
Isn't this else just an error ?

Reply:
I guess that is what we do here: 
https://github.com/ibm-openbmc/bmcweb/blob/master/redfish-core/lib/systems.hpp#L624